lua_cjson.c

```diff
@@ -698,8 +700,20 @@ static void json_append_data(lua_State *l, json_config_t *cfg,
         len = lua_array_length(l, cfg, json);
         if (len > 0 || (len == 0 && !cfg->encode_empty_table_as_object))
             json_append_array(l, cfg, current_depth, json, len);
-        else
-            json_append_object(l, cfg, current_depth, json);
+        else {
+            int as_array = 0;
+            if (lua_getmetatable(l, -1)) {
+                luaL_getmetatable(l, CJSON_AS_ARRAY_MT);
```

Comment:
@thibaultCha Now you are using a string key, "CJSON_AS_ARRAY_MT", to index the metatable, which is slow since you need to intern that C string upon every invocation of `luaL_getmetatable`. I am suggesting the following sequence:

```C
lua_pushlightuserdata(L, &some_global_c_stub_variable_in_cjson);
lua_rawget(L, LUA_REGISTRYINDEX);
```

By using the cjson table, I mean just store the metatable into the cjson module table returned by `require "cjson"`. It might be tricky in a traditional Lua C module as far as I can see. I'm also fine with using the registry table but with a lightuserdata key as demonstrated above.